tests/e2e/framework/kubernetes.go

```diff
@@ -1070,9 +1075,9 @@ func (k *KubeInfo) generateIstio(src, dst string) error {
 
 	// A very flimsy and unreliable regexp to replace delays in ingress gateway pod Spec
 	// note these config nobs aren't exposed to helm
-	content = replacePattern(content, "'10s' #connectTimeout", "'1s' #connectTimeout")
-	content = replacePattern(content, "'45s' #drainDuration", "'2s' #drainDuration")
-	content = replacePattern(content, "'1m0s' #parentShutdownDuration", "'3s' #parentShutdownDuration")
+	content = replacePattern(content, "'10s'", "'1s' #connectTimeout")
```

Comment:
I think this is a very dangerous string substitute. Maybe we should update the sources with the hash-tags?

Reply:
the whole mechanism is broken. i feel like we need knobs for control instead.

Reply:
true that. can you add TODOs with what's wrong with this code, so we remember why this code is bad.

Reply:
Added one with a link to a filed issue.